packages/x-date-pickers/src/PickersShortcuts/PickersShortcuts.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import List, { ListProps } from '@mui/material/List';
+import ListItem from '@mui/material/ListItem';
+import Chip from '@mui/material/Chip';
+import { VIEW_HEIGHT } from '../internals/constants/dimensions';
+
+interface PickersShortcutsItemGetValueParams<TValue> {
+  isValid: (value: TValue) => boolean;
+}
+
+export interface PickersShortcutsItem<TValue> {
+  label: string;
+  getValue: (params: PickersShortcutsItemGetValueParams<TValue>) => TValue;
+}
+
+export interface PickersShortcutsProps<TValue> extends Omit<ListProps, 'onChange'> {
+  /**
+   * Ordered array of shortcuts to display.
+   * If empty, does not display the shortcuts.
+   * @default `[]`
+   */
+  items?: PickersShortcutsItem<TValue>[];
+  isLandscape: boolean;
+  onChange: (newValue: TValue) => void;
+  isValid: (value: TValue) => boolean;
+}
+
+function PickersShortcuts<TValue>(props: PickersShortcutsProps<TValue>) {
+  const { items, isLandscape, onChange, isValid, ...other } = props;
+
+  if (items == null || items.length === 0) {
+    return null;
+  }
+
+  const itemsAttributs = items.map((item) => {
+    const newValue = item.getValue({ isValid });
+
+    return {
+      label: item.label,
+      onClick: () => {
+        onChange(newValue);
+      },
+      disabled: !isValid(newValue),
+    };
+  });
+
+  return (
+    <List
+      dense
+      sx={[
+        {
+          maxHeight: VIEW_HEIGHT,
+          maxWidth: 200,
+          overflow: 'hidden',
+          '&:hover': { overflow: 'auto' },
```

Comment:
This solution doesn't work on mobile. 🙈

Reply:
@alexfauquette 
I've experimented with a solution, which would be supported in mobile and I think that with only a tiny tweak it does work.
Could you have a look [at this example](https://codesandbox.io/s/confident-leftpad-htposs?file=/demo.tsx)? 🤔

Reply:
Amazing :)

I moved it into the follow-up issue, such that we can merge as it is, and add improvement after having time to think about them